src/dom/element.ts

```diff
@@ -220,15 +267,21 @@ export class Element extends Node {
   }
 
   setAttribute(name: string, value: any) {
-    this.attributes[name] = "" + value;
+    const strValue = String(value);
+    this.attributes[name] = strValue;
 
     if (name === "id") {
-      this.#currentId = value;
+      this.#currentId = strValue;
+    } else if (name === "class") {
+      this.classList._update(strValue);
     }
   }
 
   removeAttribute(name: string) {
-    this.attributes[name] = null as any as string;
+    this.attributes[name] = (null as any) as string;
```

Comment:
Why did you put parens on `(null as any)`?

Reply:
That was my IDE's formatting, I removed the changes elsewhere but missed it here.

Reply:
Yeah, change it to the coercion thanks